File: app/src/main/java/com/falcon/restaurants/common/di/presentation/PresentationComponent.java
package com.falcon.restaurants.common.di.presentation;

import com.falcon.restaurants.screens.restaurant.RestaurantsActivity;
import com.falcon.restaurants.screens.meal.MealsActivity;
import com.falcon.restaurants.screens.mealdetail.MealDetailsActivity;
import com.falcon.restaurants.screens.splash.SplashActivity;
import dagger.Subcomponent;

@PresentationScope
@Subcomponent(modules = {PresentationModule.class})
public interface PresentationComponent {

    void inject(SplashActivity activity);
    void inject(MealsActivity activity);
    void inject(RestaurantsActivity activity);
    void inject(MealDetailsActivity activity);

}

File: app/src/main/java/com/falcon/restaurants/common/di/presentation/PresentationComponent.kt
package com.falcon.restaurants.common.di.presentation
import com.falcon.restaurants.screens.restaurant.RestaurantsActivity
import com.falcon.restaurants.screens.meal.MealsActivity
import com.falcon.restaurants.screens.mealdetail.MealDetailsActivity
import com.falcon.restaurants.screens.splash.SplashActivity
import dagger.Subcomponent

@PresentationScope
@Subcomponent(modules = [PresentationModule::class])
interface PresentationComponent {

    fun inject(activity: SplashActivity)
    fun inject(activity: MealsActivity)
    fun inject(activity: RestaurantsActivity)
    fun inject(activity: MealDetailsActivity)

}
